### ISSUE TITLE
Support model values as object

### ISSUE THREAD

**Issue by ValentinH** [formly-js/angular-formly#560]
As discussed here: https://github.com/formly-js/angular-formly-website/pull/85

If I use a model like this:
```js
vm.model = {
  range: {low: 0, high: 10}
}
```
then the `vm.options.resetModel()` doesn't work. This is due to the object reference that is the same between the model and the saved version.

What do you think of using `angular.copy` during the initial value storage (only if the value is an object or an array) ?

**Comment by sasikumardr** [comment]
anothe example with ui-select. If we add any value in multi-select field, resetModel() does not clear model value (can verify this on model value json displayed below). Clicking reset again will only make model value null.

http://jsbin.com/tesagi/2/edit?html,js,output

**Comment by benoror** [comment]
@ValentinH Hi, I just tested http://jsbin.com/zoyiletefa/1/edit?js,output and **Reset** works fine, can you still replicate it?

Regarding @sasikumardr I can confirm both malfunctions. I suspect his might be a **bug related to resetModel() with Array's**

**Comment by ValentinH** [comment]
Indeed, the example you mentionned works as expected but it required quite a lot of code to display a simple range slider.

The original example (where reset does not work) is the following: http://jsbin.com/vudameneko/1/edit?js,output .
